indexd/client.py

```diff
@@ -0,0 +1,123 @@
+try:
```

Comment:
This is a direct copy of the IndexClient code, right? Can you add a file comment on both pieces so we know to change the client code when one changes.

Reply:
Some of the code, correct. Just enough for what is needed. Added note to make sure code changes are synced if necessary.

Reply:
Duplicating code like this is dangerous as they can easily get out of sync. We need to find a different solution.

What were the reasons for adding the client here?  (as opposed to other solutions for resolving the circular dependency that you may have explored)

Reply:
Was thinking about this, maybe now the import can go a single direction since you've removed import of indexclient in this repo. We can have indexclient import this client module and add whatever additional pieces it needs to function.

Reply:
The other solution was to mock out indexd (to eliminate its import in indexclient tests). After talking with Rowland this seemed to be preferable. I am not sure if we want to have to import indexd to use indexclient. Seems like at that point we should just merge it all into indexd.

Reply:
I'm guessing the indexclient isn't in here so it could be lighter weight than getting all of indexd. Is there an issue/concern with merging everything here @kulgan @philtom-ctds ? On the surface, that seems like a straightforward operation.

Reply:
The chances of being out of sync might be less than it seems at first glance. The python attributes of a Document are dynamic based on the server response, so changes would be reflected automagically. That being said, it would be cleaner to work with the json directly from a plain `requests.get()`

Reply:
I think if we drop the IndexClient/Document terminology we might see this solution a little differently. The need to call api provided by this same server could have been initially avoided by using an existing service layer logic or writing a more appropriate one instead of introducing the client library. If we had done that it might have looked similar to this PR possibly without the IndexClient/Dpcument terms. 
Whatever the server needs to do to so it returns correct result should be encapsulated within the server and I think moving the logic in here is more beneficial to that behavior rather than having it call a client library. Even though, for this particular scenario the things the server needs to do to satisfy a request are a subset of what the client already does, I don't think it couples them. The client can still add new features without this aspect of the server changing and vice versa. Also clients can be in any language, so if server changes the client should do the same to maintain compatibility. Like @ProfOak mentioned above, I would also favor working directly with json and removing the added classes

Reply:
I think @ProfOak and @kulgan make good points. Some confusion may be lifted if the didn't use classes and in particular classes of the same name as in indexclient. It's not clear to me that code syncing is actually an issue. 

I think merging indexclient into indexd (and archiving indexclient) is a less desirable solution, because the shared functionality between the two is not large and would bloat indexd.